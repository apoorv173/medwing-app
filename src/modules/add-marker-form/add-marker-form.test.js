import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import AddMarkerForm from './add-marker-form';


it('Input onchange event', () => {

    const event = {
		preventDefault:() => {},
		target: {
			name:'searchBox',
			value:''
		}
    }

	const onSubmitFn = jest.fn();
	const positionObj = {
        formatted_address: ''
    };
    
    
	const markerForm = shallow(<AddMarkerForm
            formInEditMode={true}
            onSubmit={onSubmitFn(event)}
            mapObjToEdit={positionObj}
        />
	);
    expect(markerForm.find('form').length).toEqual(1);
    expect(markerForm.find('SearchField').length).toEqual(1);
    expect(markerForm.find('button').length).toEqual(1);

	markerForm.find('form').simulate('submit', event);
    expect(onSubmitFn).toHaveBeenCalled();
});

it('renders correctly', () => {
    const positionObj = {
        formatted_address: ''
    };
  const tree = renderer
    .create(<AddMarkerForm
        formInEditMode={true}
        mapObjToEdit={positionObj}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});