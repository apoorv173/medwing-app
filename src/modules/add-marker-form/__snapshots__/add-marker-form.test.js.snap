// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <strong>
    Edit marker for 
  </strong>,
  <hr />,
  <form
    className="form-section"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="searchBox"
      >
        Enter location
      </label>
      <input
        className="form-control"
        name="searchBox"
        onChange={[Function]}
        placeholder="Please enter the location to search"
        type="text"
      />
    </div>
    <button
      className="btn btn-secondary"
      type="submit"
    >
      Edit marker
    </button>
  </form>,
]
`;
